File: Healthcare-app/android/MDBRealmDoctor/app/src/main/java/com/wekanmdb/storeinventory/model/practitioner/PractitionerRole.kt
package com.wekanmdb.storeinventory.model.practitioner

import com.wekanmdb.storeinventory.model.embededclass.Attachment
import com.wekanmdb.storeinventory.model.embededclass.Availability
import com.wekanmdb.storeinventory.model.embededclass.Codable_Concept
import com.wekanmdb.storeinventory.model.embededclass.Human_Name
import com.wekanmdb.storeinventory.model.organization.Organization
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import org.bson.types.ObjectId

open class PractitionerRole (
    @PrimaryKey var _id: ObjectId = ObjectId(),
    var identifier: String? = null,
    var active: Boolean? = null,
    var practitioner: Practitioner? = null,
    var organization: Organization? = null,
    var availableTime: Availability? = null,
    var code:Codable_Concept? = null,
    var specialty:Codable_Concept? = null
): RealmObject() {
}

File: Healthcare-app/ios/MDBRealmPatient/MDBRealmPatient/PractitionerRole.swift
//
//  PractitionerRole.swift
//  MDBRealmPatient
//
//  Created by MaDHiVaNan on 4/29/22.
//

import Foundation
import RealmSwift

class PractitionerRole: Object {
    @Persisted(primaryKey: true) var _id: ObjectId
    @Persisted var about: String?
    @Persisted var active: Bool?
    @Persisted var availableTime: Availability?
    @Persisted var code: Codable_Concept?
    @Persisted var identifier: String?
    @Persisted var name: Human_Name?
    @Persisted var organization: Organization?
    @Persisted var practitioner: Practitioner?
    @Persisted var specialty: Codable_Concept?
}



class Availability: EmbeddedObject {
@Persisted var allDay: Bool?
@Persisted var availableEndTime: String?
@Persisted var availableStartTime: String?
@Persisted var daysOfWeek: String?
}
class Practitioner: Object {
@Persisted(primaryKey: true) var _id: ObjectId
@Persisted var about: String?
@Persisted var active: Bool?
@Persisted var birthDate: Date?
@Persisted var gender: String?
@Persisted var identifier: String?
@Persisted var name: Human_Name?
@Persisted var photo: List<Attachment>
}
